src/libstd/sys/windows/fs.rs

```diff
@@ -517,6 +517,25 @@ pub fn rmdir(p: &Path) -> io::Result<()> {
     Ok(())
 }
 
+pub fn remove_dir_all(path: &Path) -> io::Result<()> {
+    for child in try!(readdir(path)) {
+        let child = try!(child).path();
+        let stat = try!(lstat(&*child));
+        if stat.data.dwFileAttributes & c::FILE_ATTRIBUTE_DIRECTORY != 0 {
```

Comment:
This logic seems quite similar to that of `FileType::new` above, could the two be consolidated?